### DISCUSSION TITLE
How can I wait for my "request" and response back to front end?

### DISCUSSION THREAD

**Discussion by YuyangZhou3** [adonisjs/core#1650] [Help] [answered]
Hi guys, I am working on the Zoom API now. I want to send my token from ZOOM API to front-end as a response. However, "token from request" is always printed first and undefined! Then the token from Zoon API" will be followed with the token. How can I make it happend? Thx!

`
const Newrequest = require("request");

class ZoomController {
  async getAccessToken({request, response, params}) {

    const instructor_id = params.id;
    try {
      let tokenRes;
      const code = request.all().code;
      console.log("the code from frontend is ", code);
      const client_id = _something_
      const client_secret = _something_
      var options = {
        method: "POST",
        url: "https://api.zoom.us/oauth/token",
        qs: {
          grant_type: "authorization_code",
          code: code,
          redirect_uri: _something_
        },
        headers: {
          Authorization:
            "Basic " +
            Buffer.from(client_id + ":" + client_secret).toString("base64")
        }
      };

      await Newrequest(options, function(error, res, body) {
        if (error) throw new Error(error);
        tokenRes = JSON.parse(body);
        console.log("token from Zoon API",tokenRes);
      });

      console.log("token from request",tokenRes);
      return response.status(200).send(tokenRes);
    } catch (error) {
      console.log(error);
      return response.status(401).send();
    }
  }
`

**Comment by RomainLanz** [comment, marked as answer]
Hey @YuyangZhou3! 👋 

Your issue is not related to AdonisJS at all.
You need first to understand how asynchronous code works in JavaScript.

To give you a hint, you are mixing `async/await` with a callback API.
I'd recommend you to read [about Promise](https://javascript.info/promise-basics) and [`async/await`](https://javascript.info/async-await).
